modules/objc/generator/gen_objc.py

```diff
@@ -780,14 +780,25 @@ def add_const(self, decl, scope=None, enumType=None): # [ "const cname", val, []
             logging.info('ignored: %s', constinfo)
         else:
             objc_type = enumType.rsplit(".", 1)[-1] if enumType else ""
-            if constinfo.classname in const_fix and objc_type in const_fix[constinfo.classname] and constinfo.name in const_fix[constinfo.classname][objc_type]:
-                fixed_const = const_fix[constinfo.classname][objc_type][constinfo.name]
-                constinfo.name = fixed_const
-                constinfo.cname = fixed_const
+            if constinfo.enumType and constinfo.classpath:
+                new_name = constinfo.classname + '_' + constinfo.name
+                const_fix.setdefault(constinfo.classpath, {}).setdefault(objc_type, {})[constinfo.name] = new_name
+                constinfo.name = new_name
+                logging.info('use outer class prefix: %s', constinfo)
+
+            if constinfo.classpath in const_fix and objc_type in const_fix[constinfo.classpath]:
+                fixed_consts = const_fix[constinfo.classpath][objc_type]
+                if constinfo.name in fixed_consts:
+                    fixed_const = fixed_consts[constinfo.name]
+                    constinfo.name = fixed_const
+                    constinfo.cname = fixed_const
+                if constinfo.value in fixed_consts:
+                    constinfo.value = fixed_consts[constinfo.value]
 
             if not self.isWrapped(constinfo.classname):
                 logging.info('class not found: %s', constinfo)
-                constinfo.name = constinfo.classname + '_' + constinfo.name
+                if not constinfo.name.startswith(constinfo.classname + "_"):
+                    constinfo.name = constinfo.classname + '_' + constinfo.name
```

Comment:
constinfo.name = constinfo.classname + '_' + constinfo.name + ' NS_SWIFT_NAME(' + constinfo.name ')'

Reply:
What is about duplicates?

Check `SIZE_256_BITS` and `SIZE_512_BITS` from 2 files.

Reply:
@alalek In Swift the values of an enum are not global - they are scoped by the enum type.
So you can have something like:
```
class Test {
  enum BiblibSize {
    case SIZE_256
    case SIZE_512
  }
  enum TeblidSize {
    case SIZE_256
    case SIZE_512
  }
  let biblibSize: BiblibSize = .SIZE_256
  let teblidSize: TeblidSize = .SIZE_256
}
```

Reply:
Updated.
Please validate with opencv_contrib.

Reply:
> Please validate with opencv_contrib.

I build with open_contrib and can confirm the enums are working well in Swift. This code works fine:
```
        let b: BeblidSize = .SIZE_256_BITS
        let t: TeblidSize = .SIZE_256_BITS
```

However I discovered the Objective-C code generation for the **aruco** module is not working (the problem is unrelated to this code change) - I will prepare a fix for that

Reply:
@komakai Thank you!